### ISSUE TITLE
Add an argument for open intervals in assert_numeric

### ISSUE THREAD

**Issue by JakubBujnowicz** [mllg/checkmate#215]
Currently `lower` acts as greater than or equal (`>=`) to `lower`. Would it be possible to add an another argument that would act as `>`? Or maybe something like boolean `lower.eq` that would decide whether equality is allowed.

Obviously the same goes for `upper`. 

One can use a trick with the machine epsilon, i.e. `assert_numeric(0, lower = .Machine$double.eps)`, but neither the code nor the error message (`Error: Assertion on '0' failed: Element 1 is not >= 2.22045e-016.`) is elegant. The same trick used for something other than zero may result in an even more confusing error message, i.e. 
```assert_numeric(1, lower = 1 + .Machine$double.eps)``` 
leading to `Error: Assertion on '1' failed: Element 1 is not >= 1.`.

**Comment by mllg** [comment]
Yes, that is possible. However, I am currently busy with other projects, so it will take a while.

**Comment by JakubBujnowicz** [comment]
That's great, thanks!

**Comment by szimmer** [comment]
Yes, this is particularly common when one needs a non-negative number. 0 shouldn't be allowed but a small but positive number would be and it's hard to specify as-is.
